functions-python/update_validation_report/src/main.py

```
@@ -0,0 +1,189 @@
+
+    session = start_db_session(os.getenv("FEEDS_DATABASE_URL"), echo=False)
+    latest_datasets = get_latest_datasets(session)
+    logging.info(f"Retrieved {len(latest_datasets)} latest datasets.")
+
+    dataset_blobs = get_dataset_blobs_for_validation(
+        bucket_name, validator_version, latest_datasets
+    )
+    logging.info(f"Retrieved {len(dataset_blobs)} blobs to update.")
+
+    updated_datasets = update_dataset_metadata(dataset_blobs, validator_version)
+    response = {
+        "message": f"Updated {len(updated_datasets)} validation report(s).",
+        "updated_datasets": sorted(updated_datasets),
+        "ignored_datasets": sorted(
+            [
+                dataset_id
+                for _, dataset_id in latest_datasets
+                if dataset_id not in updated_datasets
+            ]
+        ),
+    }
+    return response, 200
+
+
+def get_validator_version():
+    """
+    Get the version of the validator
+    :return: the version of the validator
+    """
+    web_validator_endpoint = os.getenv("WEB_VALIDATOR_URL")
+    response = requests.get(f"{web_validator_endpoint}/version")
+    validator_version = response.json()["version"]
+    logging.info(f"Validator version: {validator_version}")
+    return validator_version
+
+
+def update_dataset_metadata(
+    dataset_blobs: List[storage.Blob], validator_version: str
+) -> List[str]:
+    """
+    Update the metadata of the dataset blobs - This will trigger the validation report processor
+    :param dataset_blobs: The dataset blobs to update
+    :param validator_version: The version of the validator
+    :return: The list of updated datasets stable ids
+    """
+    max_retry = os.getenv("MAX_RETRY", None)
+    batch_size = int(os.getenv("BATCH_SIZE", 5))
+    sleep_time = int(os.getenv("SLEEP_TIME", 5))
+    logging.info(f"Max retry: {max_retry}, Batch size: {batch_size}")
+
+    batch_count = 0
+    updated_datasets = []
+    for blob in dataset_blobs:
```

[suggestion]: The loop does not handle exceptions. I suggest wrapping a single dataset's processing within try/catch and ignoring the errors. This way, we can continue processing even if one or a few operations fail.

---

done ✅  Added a `try/catch` clause for each loop.